assets/model_monitoring/components/src/model_data_collector_preprocessor/run.py

```diff
@@ -167,6 +167,15 @@ def _extract_data_and_correlation_id(df: DataFrame, extract_correlation_id: bool
     otherwise, return the dataref content which is a url to the json file.
     """
 
+    def convert_object_to_str(dataframe):
+        columns = dataframe.columns
+        index = 0
+        for t in dataframe.dtypes:
```

Comment:
nit: can use this to get the index in the for loop

for index, t in enumerate(dataframe.dtypes):